scripts/quests/otherAreas/The_Sand_Charm.lua

```diff
@@ -0,0 +1,143 @@
+-----------------------------------
+-- The Sand Charm
+-----------------------------------
+-- Log ID: 4, Quest ID: 8
+-- !addquest 4 8
+-- Blandine  : !pos 23 -7 41 249
+-- Zexu      : !pos 31.511 -9.001 23.496 249
+-- Celestina : !pos -37.624 -16.050 75.681 249
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.THE_SAND_CHARM)
+
+quest.reward =
+{
+    exp      = 2000,
+    gil      = 2000,
+    ki       = xi.ki.MAP_OF_BOSTAUNIEUX_OUBLIETTE,
+    fameArea = xi.fameArea.WINDURST,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+                player:getFameLevel(xi.fameArea.WINDURST) >= 4 and
+                xi.settings.map.FISHING_ENABLE == true
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Blandine'] =
+            {
+                onTrigger = function(player, npc)
+                    local xPos = player:getXPos()
+                    local zPos = player:getZPos()
+
+                    if zPos <= 29 or zPos >= 38 or xPos <= 16 or xPos >= 32 then
+                        if quest:getVar(player, 'Prog') == 0 then
+                            return quest:progressEvent(125) -- I know he's out there
+                        elseif quest:getVar(player, 'Prog') == 2 then
+                            return quest:progressEvent(124) -- Celestina doesn't want to hear it anymore
+                        end
+                    end
+                end,
+            },
+
+            ['Zexu'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(123) -- Word is, pirates got 'im
```

Comment:
I think the little text excerpts are quite helpful actually, big fan

Reply:
I'll add a few more when I get home

Reply:
All set